### ISSUE TITLE
[Adjust Rule] SIM300 false positive with literals on both sides

### ISSUE THREAD

**Issue by ghost** [MartinThoma/flake8-simplify#155]
## Desired change

* **Rule(s)**: SIM300
* **Adjustment**: Don't trigger if both sides of a comparison are literals

## Explanation

These comparisons are pointless, but swapping them around (only to be told to swap them around again after) is also pointless.

### Example 1

```py
if "a" == "b":
    pass
```
```
SIM300: Use '"b" == "a"' instead of '"a" == "b"' (Yoda-conditions)
```

### Example 2

```py
if 1 == 2:
    pass
```
```
SIM300: Use '2 == 1' instead of '1 == 2' (Yoda-conditions)
```

**Comment by MartinThoma** [comment]
Hm... Although the error message is misleading, I would still say that such code should be fixed. Instead of comparing constants, I'd recommend to use the result of the comparison directly. Or do you see a case where you would want to keep the constant comparison?
